addons/payment_buckaroo/models/payment_transaction.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from werkzeug import urls
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+from odoo.addons.payment_buckaroo.controllers.main import BuckarooController
+
+_logger = logging.getLogger(__name__)
+
+
+def _normalize_keys_upper(data):
+    """ Set all keys of a dictionary to uppercase.
+
+    As Buckaroo parameters names are case insensitive, we can convert everything to upper case to
+    easily detected the presence of a parameter by checking the uppercase key only.
+
+    :param dict data: The dictionary whose keys must be set to uppercase
+    :return: A copy of the original data with all keys set to uppercase
+    :rtype: dict
+    """
+    return {key.upper(): val for key, val in data.items()}
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    # Buckaroo status  TODO ANV const
+    _buckaroo_valid_tx_status = [190]
+    _buckaroo_pending_tx_status = [790, 791, 792, 793]
+    _buckaroo_cancel_tx_status = [890, 891]
+    _buckaroo_error_tx_status = [490, 491, 492]
+    _buckaroo_reject_tx_status = [690]
+
+    def _get_specific_rendering_values(self, processing_values):
+        if self.provider != 'buckaroo':
+            return super()._get_specific_rendering_values(processing_values)
+
+        buckaroo_tx_values = dict(processing_values)
+        base_url = self.acquirer_id._get_base_url()
+        buckaroo_tx_values.update({
+            'Brq_websitekey': self.acquirer_id.buckaroo_website_key,
+            'Brq_amount': processing_values['amount'],
+            'Brq_currency': self.currency_id.name or '',
```

Comment:
The `name` is required on `res.currency` so I guess that we can safely do:
```suggestion
            'Brq_currency': self.currency_id.name,
```